mdpow/forcefields.py

```diff
@@ -180,163 +229,228 @@ def get_water_model(watermodel=DEFAULT_WATER_MODEL):
-        identifier="toluene", itp="1tol.itp", coordinates="1tol_amber.gro"
-    ),
+ALL_FORCEFIELDS: Dict[str, Forcefield] = {
+    "OPLS-AA": OPLS_AA,
+    "CHARMM": CHARMM,
+    "AMBER": AMBER,
 }
 
 #: Solvents available in GROMACS; the keys of the dictionary
 #: are the forcefields.
 GROMACS_SOLVENT_MODELS = {
-    "OPLS-AA": OPLS_SOLVENT_MODELS,
-    "CHARMM": CHARMM_SOLVENT_MODELS,
-    "AMBER": AMBER_SOLVENT_MODELS,
+    "OPLS-AA": OPLS_AA.solvent_models,
+    "CHARMM": CHARMM.solvent_models,
+    "AMBER": AMBER.solvent_models,
 }
 
 
-def get_solvent_identifier(solvent_type, model=None, forcefield="OPLS-AA"):
+def _get_forcefield(ff_name: str) -> Forcefield:
+    """Get the :class:`Forcefield` instance corresponding to a given name."""
+    try:
+        return ALL_FORCEFIELDS[ff_name]
+    except KeyError:
+        raise ValueError(f"Forcefield `{ff_name}` not found.")
+
+
+def get_solvent_identifier(
+    solvent_type, model=None, forcefield: Union[Forcefield, str] = OPLS_AA
+):
     """Get the identifier for a solvent model.
 
     The identifier is needed to access a water model (i.e., a
-    :class:`GromacsSolventModel`) through
-    :func:`get_solvent_model`. Because we have multiple water models
-    but only limited other solvents, the organization of these models
-    is a bit convoluted and it is best to obtain the desired water
-    model in these two steps::
+    :class:`GromacsSolventModel`) through :func:`get_solvent_model`. Because we
+    have multiple water models but only limited other solvents, the organization
+    of these models is a bit convoluted and it is best to obtain the desired
+    water model in these two steps::
+
+      identifier = get_solvent_identifier("water", model="tip3p") model =
+      get_solvent_model(identifier)
 
-      identifier = get_solvent_identifier("water", model="tip3p")
-      model = get_solvent_model(identifier)
 
+    For ``solvent_type`` *water*: either provide ``None`` or "water" for the
+    specific ``model`` (and the default water model for the :class:`Forcefield`
+    will be selected, or a specific water model such as "tip3p" or "spce" (see
+    :data:`GROMACS_WATER_MODELS`). For other "octanol" or "wetoctanol" of
+    OPLS-AA forcefield, the ``model`` is used to select a specific model. For
+    other solvents and forcefields, "model" is not required.
 
-    For ``solvent_type`` *water*: either provide ``None`` or "water"
-    for the specific ``model`` (and the default
-    :data:`DEFAULT_WATER_MODEL` will be selected, or a specific water
-    model such as "tip3p" or "spce" (see
-    :data:`GROMACS_WATER_MODELS`). For other "octanol" or "wetoctanol"
-    of OPLS-AA forcefield, the ``model`` is used to select a specific
-    model. For other solvents and forcefields, "model" is not required.
+    :Raises ValueError: If there is no identifier found for the combination.
 
-    :Returns: Either an identifier or ``None``
+    :Returns: An identifier
```

Comment:
add a line
```reST

.. versionchanged:: 0.9.0
   Raises :exc:`ValueError` instead of returning ``None``.
```

Also note this change of behavior in CHANGES.